framework_tvb/tvb/adapters/simulator/simulator_fragments.py

```diff
@@ -138,41 +138,36 @@ def prepare_stimulus_fragment(simulator, rendering_rules, is_surface_simulation,
 class SimulatorModelFragment(ABCAdapterForm):
     def __init__(self):
         super(SimulatorModelFragment, self).__init__()
-        self.model_choices = get_ui_name_to_model()
-        default_model = list(self.model_choices.values())[0]
+        default_model = ModelsEnum.GENERIC_2D_OSCILLATOR
 
         self.model = SelectField(
-            Attr(Model, default=default_model, label=SimulatorAdapterModel.model.label,
-                 doc=SimulatorAdapterModel.model.doc), choices=self.model_choices, name='model')
+            EnumAttr(default=default_model, label=SimulatorAdapterModel.model.label,
+                     doc=SimulatorAdapterModel.model.doc), name='model')
 
     def fill_from_trait(self, trait):
         # type: (SimulatorAdapterModel) -> None
-        self.model.data = trait.model.__class__
+        self.model.data = type(trait.model)
 
     def fill_trait(self, datatype):
-        if type(datatype.model) != self.model.value:
-            datatype.model = self.model.value()
+        if type(datatype.model) != type(self.model.value.instance):
```

Comment:
Why not avoid creating an instance and use .value instead? I think I saw this contruction in more places.

Reply:
Yeah, you are right, but I think this is the only place where I didn't do like you say.